insights/parsers/kpatch_patches.py

```diff
@@ -0,0 +1,40 @@
+"""
+KpatchPatches - report locally stored kpatch patches
+====================================================
+
+This parser creates a list of the module names of locally
+stored kpatch modules. If no modules are installed, the
+list will be empty.
+
+"""
+
+from .. import parser, Parser
+from insights.specs import Specs
+import re
+
+
+@parser(Specs.kpatch_patch_files)
+class KpatchPatches(Parser):
+    """
+    A parser for getting locally stored kpatch-patch modules
+    """
+    __re_cannot_access = re.compile(r"No such file or directory$")
+    __re_archive = re.compile(r"\.xz$")
+
+    def get_patches(self):
+        return self.patches
+
+    def parse_content(self, content):
+        self.patches = content
+        # If kpatch is not installed, the directory will not exist.
+        # In that case, the list of local kpatch files is empty.
+        if self.__re_cannot_access.search(content[0]):
+            self.patches = []
+            return
+
+        # remove module archives, convert dashes to underscores, remove ".ko" suffixes
+        self.patches =  \
```

Comment:
Comparing the multiple `lambda` functions, how do you think split it as below?
```
self.patches = [p.split('.')[0].replace(...) for p in self.patches]
```

Reply:
I simplify the parsing greatly using your suggestion in the latest patch